crates/bevy_app/src/app.rs

```diff
@@ -30,21 +32,36 @@ use bevy_utils::tracing::info_span;
 /// }
 /// ```
 pub struct App {
-    pub world: World,
-    pub runner: Box<dyn Fn(App)>,
-    pub schedule: Schedule,
+    pub world: ManuallyDrop<World>,
```

Comment:
This is a drop in code clarity / friendliness for a relatively "core" type that I expect people to look at on a regular basis (and also an increase in unsafe).

Can we discuss a bit what this ~17k wins us? If this were optimizing for web/wasm deployments (which have tight network constraints / lots of people that care about load times), that would make more sense to me. But this is specifically optimizing dynamically linked outputs, which generally won't be published.

This change also only affects "top level" apps, so this won't come into play for "dynamically linked plugins" if we decide to go that route in the future.

So I think the intent here is to optimize "iterative compile times" when using `--features dynamic`. If so, it would be good to measure how much we're winning here and weight that against the negatives pointed out above.

Reply:
> So I think the intent here is to optimize "iterative compile times" when using `--features dynamic`.

Mostly correct, though it should also save some time without `--features dynamic` too. Before this PR the drop glue would be generated both in the game (for unwinding on panics) and in bevy_app (for dropping at the end of `AppBuilder::run`). After this PR it is only generated in bevy_app.